store/src/types.rs

```diff
@@ -193,21 +193,37 @@ impl AppendOnlyFile {
 	pub fn flush(&mut self) -> io::Result<()> {
 		if self.buffer_start_bak > 0 {
 			// Flushing a rewound state, we need to truncate via set_len() before applying.
-			self.file.set_len(self.buffer_start as u64)?;
+			// Drop and recreate, or windows throws an access error
+			self.mmap = None;
+			self.file = None;
+			{
+				let file = OpenOptions::new()
+					.read(true)
+					.create(true)
+					.write(true)
+					.open(&self.path)?;
+				file.set_len(self.buffer_start as u64)?;
+			}
+			let file = OpenOptions::new()
+				.read(true)
+				.create(true)
+				.append(true)
+				.open(&self.path)?;
```

Comment:
File permissions for `set_len` on windows need to be as set here or it fails. Hence the drop and reopen with original permissions after the `set_len`.